Library/Homebrew/dependable.rb

```diff
@@ -9,7 +9,7 @@
 module Dependable
   # `:run` and `:linked` are no longer used but keep them here to avoid their
   # misuse in future.
-  RESERVED_TAGS = [:build, :optional, :recommended, :run, :test, :linked].freeze
+  RESERVED_TAGS = [:build, :optional, :recommended, :run, :test, :linked, :auto].freeze
```

Comment:
I'm fine with whatever naming, though `if_needed` to me implies that whether it's needed or not is yet to be determined which is not true - the dependency is created _after_ this has been determined (for the currently running system).

Alternative names could be something like `:implicit` etc. Idea is that these are filtered from the output JSON and only explicitly-defined `depends_on` are included.

Reply:
@Bo98 `only_if_needed`, `only_if_missing`, `was_missing`, `not_whiched` maybe? How is this consumed? Dependencies always created as normal and with these tags and then omitted from JSON?

Reply:
> How is this consumed? Dependencies always created as normal and with these tags and then omitted from JSON?

Taking the example of URL extension/archive dependencies, the flow is:

* Call `url "https://.../file.zst"`
* The software spec create a resource with this url
* The software spec passes the resource to `DependencyCollector#add`
* The dependency collector will read `.zst` and check `which("zstd")`
* If zstd is not available, add `Dependency.new("zstd", [:build, :test:, :auto])` to the list of dependencies
* The formula installer does not consider `:auto` at all. It has no meaning there. It will install all dependencies in the list - its addition to the dep list is already an indication that it is needed.
* JSON generation will do `reject(&:auto?)` when collecting the list of dependencies. This is the only place where the new tag is used.